src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,675 @@
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return iterable
+     */
+    protected function getPropertiesBySchema(Schema $schema): iterable
+    {
+        if (isset($schema->properties)) {
+            return $schema->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     *
+     * @return iterable
+     */
+    protected function getPropertiesByReference(Reference $schema): iterable
+    {
+        if (isset($schema->properties)) {
+            return $schema->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return array
+     */
+    protected function requestBodyParserBySchema(Schema $schema): array
+    {
+        $response = [];
+        foreach ($schema->properties as $key => $schemaObject) {
+            if ($schemaObject instanceof Schema && isset(($schemaObject->properties)) && !empty($schemaObject->properties)) {
+                return $this->requestBodyParserBySchema($schemaObject);
+            }
+            if ($schemaObject instanceof Reference && isset(($schemaObject->properties)) && !empty($schemaObject->properties)) {
+                return $this->requestBodyParserByReference($schemaObject);
+            }
+        }
+
+        foreach ($schema->properties as $key => $schemaObject) {
+            if (isset($schemaObject->type) && $schemaObject->type !== 'array') {
+                $response[$key] = $schemaObject->type;
+
+                continue;
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if (isset($schemaObject->items->properties) && isset($schemaObject->items->properties['type'])) {
+                $response[$key] = 'array[]:' . $schemaObject->items->properties['type']->type;
+
+                continue;
+            }
+            if (isset($schemaObject->items->type)) {
+                $response[$key] = 'array[]:' . $schemaObject->items->type;
+
+                continue;
+            }
+            $pathUsingSchemaObject = '';
+            if ($schemaObject->items instanceof Schema) {
+                $pathUsingSchemaObject = $this->getPathUsingSchema($schemaObject->items);
+            }
+            if ($schemaObject->items instanceof Reference) {
+                $pathUsingSchemaObject = $this->getPathUsingReference($schemaObject->items);
+            }
+            $response[$this->getInflectorFactory()->camelize($this->getPathUsingSchema($schema))] = $pathUsingSchemaObject . '[]:' . $this->getInflectorFactory()->camelize($pathUsingSchemaObject);
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     *
+     * @return array
+     */
+    protected function requestBodyParserByReference(Reference $schema): array
```

Comment:
The method name is not clear to me. Please also extract methods as explained above.